[Tien Yu] iP

tienyu2000 opened the pull request:
#### DukePro
“Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))

DukePro frees your mind of having to remember things you need to do. It's,

* text-based
* easy to learn
* ~~FAST~~ SUPER FAST to use

All you need to do is,

1. download it from [here](https://nus-cs2103-ay2223s1.github.io/website/schedule/week4/project.html).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

And it is **FREE!**

Features:

 - [x] Managing tasks
 - [ ] Managing deadlines (coming soon)
 - [ ] Reminders (coming soon)

Commit:
Add Gradle support

tienyu2000 committed:
Level-1

tienyu2000 committed:
Level-2

tienyu2000 committed:
Level-3

tienyu2000 committed:
Level-4

tienyu2000 committed:
Added inputs and expected

tienyu2000 committed:
Level-5

tienyu2000 committed:
Level-6

tienyu2000 committed:
Level7

tienyu2000 committed:
Level8

tienyu2000 committed:
Merge branch branch-Level-7

tienyu2000 committed:
Merge branch branch-Level-8

sltsheryl commented on src/main/java/Deadline.java:
```diff
@@ -0,0 +1,13 @@
+public class Deadline extends Task {
```
Missing JavaDocs which is a bit overdue for this week

sltsheryl commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,34 @@
+public class Task {
+    protected boolean isDone;
```
Good naming to show it's a boolean 👍🏻

sltsheryl commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,34 @@
+    protected String description;
+    protected int taskNumber;
+
+    public Task(String desc) {
```
The short form desc could be abit misleading, could be descending

sltsheryl commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,67 @@
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
+        System.out.println("Hello! I'm still evolving!\nWhat can I do for you, my highness?");
+        Scanner in = new Scanner(System.in);
```
Should incorporate all the code in OOP style in their individual classes

sltsheryl commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,67 @@
+                    break;
+                } else if (input.equals("list")) {
+                    System.out.println("Here are the tasks in your list:");
+                    for (int i = 1; i <= taskList.size(); i++) {
```
Can do abstraction, instead of doing a for loop here, could have a printTaskList() method

sltsheryl commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,67 @@
+                    System.out.println("Noted. I've removed this task:\n" + currTask.toString() +
+                            "\nNow you have " + taskList.size() + " tasks left in the list.");
+                } else {
+                    if (input.equals("todo") || input.equals("deadline") || input.equals("event")) {
```
Could try using switch statements

sltsheryl reviewed (commented):
I think week 3 hasn't really been implemented yet. I can help to give you more feedback when you're done :) All the best!

tienyu2000 committed:
Refactor using OOP

tienyu2000 committed:
Use Packages

tienyu2000 committed:
Merge add-gradle-support

tienyu2000 committed:
Add gradle support

tienyu2000 committed:
Add JUnit Tests

tienyu2000 committed:
Added Javadocs

tienyu2000 committed:
Tweak code to comply with coding standard

tienyu2000 committed:
Level9

tienyu2000 committed:
Merging branch-A-JavaDoc with master

tienyu2000 committed:
Merge branch-A-CodingStandard with master

tienyu2000 committed:
Merge branch-Level-9 with master

sltsheryl commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,88 @@
+    }
+
+    /**
+     * Run method to execute the Duke instance.
```
If you see the general format of javadocs for method, they usually write in present tense and verb 
ie. Executes the Duke instance

sltsheryl commented on src/main/java/duke/handlers/DeadlineHandler.java:
```diff
@@ -0,0 +1,28 @@
+
+public class DeadlineHandler {
+    /**
+     * Handles the DEADLINE command.
```
Clear documentation!

sltsheryl commented on src/main/java/duke/models/FormatDate.java:
```diff
@@ -0,0 +1,23 @@
+            .appendOptional(DateTimeFormatter.ofPattern("yyyy/MM/dd"))
+            .toFormatter();
+
+    public FormatDate(String date) {
```
Missing javadocs!

sltsheryl commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,88 @@
+import java.util.Scanner;
+import static duke.services.Ui.dukePrint;
+
+public class Duke {
```
Each class should have a Javadocs as well

sltsheryl commented on src/main/java/duke/services/Ui.java:
```diff
@@ -0,0 +1,17 @@
+import duke.exceptions.DukeException;
+
+public class Ui {
+    public static void dukePrint(String str) {
```
Could be clearer in method naming? Specify what you are printing

sltsheryl commented on src/main/java/duke/models/Deadline.java:
```diff
@@ -0,0 +1,20 @@
+public class Deadline extends Task {
+    protected FormatDate date;
+
+    public Deadline(String description, String by) {
```
Could add Javadocs for constructor

sltsheryl reviewed (commented):
Overall, the documentation is there 😁 The classes are implemented clearly :) But there's missing Javadocs for each class (above class name) + some methods!

tienyu2000 committed:
Added Checkstyle

tienyu2000 committed:
Level10

tienyu2000 committed:
Merge Checkstyle with master branch

tienyu2000 committed:
Merge Level-10 with master branch

tienyu2000 committed:
Added assertion statements to various methods to signify important assumptions.

Some important assumptions are not stated clearly in code and cannot be identified easily.

Placed some assertion statements to improve code readability to allow users to understand some important assumptions.

tienyu2000 committed:
Improve code quality and readability.

Renamed package names such that they are all in lower case.

Edited method header comments to fit the coding standard.

tienyu2000 committed:
Merge pull request #2 from tienyu2000/branch-A-Assertions

Added assertion statements to various methods to signify important as…

tienyu2000 committed:
Merge branch-A-CodeQuality with master branch

tienyu2000 committed:
Added Extension B-Snooze Command.

Snooze command allows user to snooze an event or deadline task to another date.

Snooze command only works for event or deadline tasks, and not for todos.

Snooze command will prompt user to input in the corresponding format if the initial input is not in the correct format.

tienyu2000 committed:
Fixed debug issues

tienyu2000 committed:
Added a representative screenshot

tienyu2000 committed:
Added user guide

tienyu2000 committed:
Fixed minor bug issues

tienyu2000 committed:
Fixed minor issue

tienyu2000 committed:
Rename file

tienyu2000 committed:
Debugged a syntax error

tienyu2000 committed:
Deleted UI.png

tienyu2000 committed:
Added Ui.png

tienyu2000 committed:
Test alert box
